source/docs/software/basic-programming/git-getting-started.rst

```diff
@@ -0,0 +1,250 @@
+Git Version Control Introduction
+================================
+
+`Git <https://git-scm.com/about>`_ is a Distributed Version Control System (VCS) created by Linus Torvalds, also known for creating and maintaining the linux kernel. Version Control is a system for tracking changes of code for developers. The advantages of Git Version Control are:
+
+- Seperate testing environments into *branches*
+- Ability to navigate to a particular *commit* without removing history
+- Ability to manage *commits* in various ways, including combining them
+- Various other features, see `here <https://git-scm.com/about>`__
+
+Prerequisites
+-------------
+
+You have to download and install Git from the following links:
+
+- `Windows <https://git-scm.com/download/win>`_
+- `Mac OS X <https://git-scm.com/download/mac>`_
+- `Linux <https://git-scm.com/download/linux>`_
+
+.. note:: You may need to add Git to your path
+
+Git Vocabulary
+--------------
+
+Git is composed of several primary elements:
+
+- *Repository:* the actual local of your code, including a ``.git`` folder in the root directory
+- *Commits:* a particular saved state of the repository, this includes all files and additions
+- *Branches:* a means of separating various commits, having a unique history. This is primarily used for separating development and stable branches.
+- *Push:* update the remote repository with your local changes
+- *Pull:* update your local repository with the remote changes
+- *Cloning:* retrieving a local copy of a repository to modify
+- *Forking:* duplicating a pre-existing repository to modify, and to compare against the original
+- *Merging:* combining various changes from different branches/commits/forks into a single history
+
+Repository
+----------
+
+A Git repository is a data structure containing the structure, history, and files of a project.
+
+Git repositories consist of:
+
+- A ``.git`` folder. This folder contains the various information about the repository. 
+- A ``.gitignore`` file. This file contains the files or directories that you do *not* want included when you commit.
+- Files and folders. This is the main content of the repository.
+
+Creating the repository
+^^^^^^^^^^^^^^^^^^^^^^^
+
+You can store the repository locally, or through the cloud. `Github <https://github.com/>`_ is a popular free hosting service. Numerous developers use it, and that's what this tutorial will use.
```

Comment:
Locally ~or~ and in the cloud

Reply:
Might be better to say locally or remotely, which could be the cloud, but also on other medium like a flash drive (which is useful to know for using git at competitions)